libs/ledger-live-common/src/families/icon/api/index.ts

```diff
@@ -0,0 +1,181 @@
+
+  return data;
+}
+
+export const getAccount = async (
+  addr: string,
+  currency: CryptoCurrency,
+): Promise<AccountType | null> => {
+  try {
+    const query = querystring.stringify({
+      address: addr,
+    });
+    const data = await fetch(`${getApiUrl(currency)}/addresses/details/${addr}?${query}`);
+    return data;
+  } catch (error) {
+    return null;
+  }
+};
+
+export const getCurrentBlockHeight = async (currency: CryptoCurrency): Promise<number> => {
+  const data = await fetch(`${getApiUrl(currency)}/blocks`);
+  return data?.[0].number;
+};
+
+/**
+ * Returns true if account is the signer
+ */
+function isSender(transaction: IconTransactionType, addr: string): boolean {
+  return transaction.from_address === addr;
+}
+
+/**
+ * Map transaction to an Operation Type
+ */
+function getOperationType(transaction: IconTransactionType, addr: string): OperationType {
+  return isSender(transaction, addr) ? "OUT" : "IN";
+}
+
+/**
+ * Map transaction to a correct Operation Value (affecting account balance)
+ */
+function getOperationValue(transaction: IconTransactionType, addr: string): BigNumber {
+  if (isSender(transaction, addr)) {
+    return transaction.value
+      ? new BigNumber(transaction.value).plus(transaction.transaction_fee)
+      : new BigNumber(0);
+  } else {
+    return transaction.value ? new BigNumber(transaction.value) : new BigNumber(0);
+  }
+}
+
+/**
+ * Map the ICON history transaction to a Ledger Live Operation
+ */
+function txToOperation(
+  accountId: string,
+  addr: string,
+  transaction: IconTransactionType,
+): IconOperation {
+  const type = getOperationType(transaction, addr);
+
+  return {
+    id: encodeOperationId(accountId, transaction.hash, type),
+    accountId,
+    fee: new BigNumber(transaction.transaction_fee),
+    value: getOperationValue(transaction, addr),
+    type,
+    hash: transaction.hash,
+    blockHash: null,
+    blockHeight: transaction.block_number,
+    date: new Date(transaction.block_timestamp / 1000),
+    senders: [transaction.from_address],
+    recipients: [transaction.to_address],
+    extra: {},
+    hasFailed: transaction.status !== "0x1",
+  };
+}
+
+/**
+ * Fetch operation list
+ */
+export const getOperations = async (
+  accountId: string,
+  addr: string,
+  skip: number,
+  currency: CryptoCurrency,
+): Promise<Operation[]> => {
+  return await fetchOperationList(accountId, addr, skip, currency);
+};
+
+export const fetchOperationList = async (
+  accountId: string,
+  addr: string,
+  skip: number,
+  currency: CryptoCurrency,
+  prevOperations: IconOperation[] = [],
+): Promise<IconOperation[]> => {
+  const data = await getTxHistory(addr, skip, currency);
+  const operations = data.map((transaction: IconTransactionType) =>
+    txToOperation(accountId, addr, transaction),
+  );
+
+  const mergedOp = [...prevOperations, ...operations];
+  if (operations.length < LIMIT) {
+    return mergedOp.filter(Boolean).sort((a: any, b: any) => b.date - a.date);
```

Comment:
If I understood well, shouldn't we do this once instead of recursively ? after the main getOperations method call ?